### PR TITLE
Add v3 of our proto

### PR DESCRIPTION
**Pull request by RichiH**
Also see https://docs.google.com/document/d/1NryqxXkWM5a-B538r7JmramirO-3PxPavnqYPKINjRo/edit?ts=5a26dff9#

### CODE REVIEW

File: proto [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+message Timeseries {
+  repeated Label label;  // apply to all points.
+  // If there is more than one point, all must have a timestamp field
+  // and they must be in increasing order of timestamp.
+  repeated Point point;
+}
+
+message MetricDescriptor {
+  required string name;
```
Comment (brian-brazil) [inline proto]: Mention that names must be unique within a MetricSet

File: proto [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    StateSetValue state_set_value;
+    HistogramValue histogram_value;
+    SummaryValue summary_value;
+    InfoValue info_value;
+  }
+  // Iff COUNTER or CUMULATIVE_HISTOGRAM type. The cumulative value is over
+  // the time interval (start_timestamp, timestamp] where both timestamps
+  // are nanoseconds since the epoch.
+  // If start_timestamp is present and timestamp is absent,
+  // the backend assigned timestamp may be used to construct the time
```
Comment (brian-brazil) [inline proto]: the word "backend" might need to be carefully defined, if I were to look at this in isolation I wouldn't be sure if this was the instrumentation client or the motioning system.

File: proto [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  }
+}
+
+message InfoValue {
+  repeated Label info;
+}
+
+
+message HistogramValue {
+  // Equivalent wrt precision issues in comparison with mean as discussed in
```
Comment (brian-brazil) [inline proto]: We don't need this comment here

File: proto [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+message SummaryValue {
+  // Sum and count are optional since some systems (e.g. DropWizard) don’t
+  // expose them. The start_timestamp only applies to the sum and count.
+  // The quantiles can be reset at arbitrary unknown times.
+  optional double sum;
+  optional int64 count;
+  repeated Quantile quantile;
+
+  message Quantile {
+    required double quantile;  // must be in the interval (0.0, 1.0]
```
Comment (brian-brazil) [inline proto]: 0 <= quantile <= 1 would be clearer (0 is fine if 1 is).

File: proto [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+    // An optional example of a value that fell into this bucket.
+    optional Exemplar exemplar;
+  }
+  message Exemplar {
+    // The value of the example.
+    required double value;
+
+    // The timestamp of the example. Nanoseconds since the epoch.
+    required int64 timestamp;
```
Comment (brian-brazil) [inline proto]: Hmm, does this have to be required?

**Comment by RichiH** [comment]
@brian-brazil Do you want to push an amendment commit into this PR branch?

**Comment by brian-brazil** [comment]
I'm pretty busy at the moment, only the last point requires broader discussion.

**Comment by RichiH** [comment]
Updated. We need to discuss your last point tomorrow, then.

File: proto [hunk 6]
```diff
@@ -0,0 +1,213 @@
+// Proto2 syntax (no tag numbers). Interpret "required" as
```
Comment (lucperkins) [inline proto]: Could you name this file `SOME-FILENAME.proto` ? That way we can get proper syntax highlighting on GitHub.

**Comment by brian-brazil** [comment]
This is obselete.

**Pull request closed without merging**